opencog/atoms/bind/PatternLink.cc

```diff
@@ -549,50 +582,50 @@ void PatternLink::unbundle_virtual(const std::set<Handle>& vars,
 			_pat.executable_terms.insert(sh);
 			_pat.executable_holders.insert(sh);
 			add_to_map(_pat.in_executable, sh, sh);
-			// But they're virtual only if they have two or more
-			// unquoted, bound variables in them. Otherwise, they
-			// can be evaluated on the spot.
 			if (2 <= num_unquoted_in_tree(sh, vars))
-			{
-				is_virtual = true;
 				is_black = true;
-			}
 		}
 		for (const Handle& sh : feol.holders)
 			_pat.executable_holders.insert(sh);
 
-		if (is_virtual)
-			virtual_clauses.push_back(clause);
-		else
-			fixed_clauses.push_back(clause);
+		virtual_clauses.push_back(clause);
```

Comment:
Not all are virtual.  Some are marked as "fixed" on line https://github.com/opencog/atomspace/pull/161/files#diff-aafb2f2e1b6e1603ab489430cefcc6fbR526